### PR TITLE
Imputeme

### PR DESCRIPTION
**Pull request by lassefolkersen**
## PR checklist

First PR on this, as recommended by @grst 

* [x] It's pulling directly from a personal quay.io address, because I had to make some changes in the impute-me software code, and I want to postpone a next release (and consequent [biocontainers update, the current is v1.0.4](https://biocontainers.pro/tools/imputeme)) untill all these small things are fixed.
* [x] It's producing the right json file that is the main output, but it still gives some error - shouldn't the output file be copied to the work-dir by [this command](https://github.com/lassefolkersen/modules/blob/e6df581843f271218009e58acccfccce0cea10cc/modules/imputeme/vcftoprs/main.nf#L60)?
* [x] Not really working with singularity at all, I've not even tried yet - @maxulysse recommend writing to biocontainers directly, it could also be possible to 'just let NF do it's thing' (do the conversion itself)

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [x] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [x] If necessary, include test data in your PR.
- [x] Remove all TODO statements.
- [x] Emit the `versions.yml` file.
- [x] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [x] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
    - [x] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [x] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd`
    - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd`

### CODE REVIEW

File: modules/imputeme/vcftoprs/main.nf [hunk 1]
```diff
@@ -0,0 +1,67 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+
+params.options = [:]
+options        = initOptions(params.options)
+
+process IMPUTEME_VCFTOPRS {
+    tag "$meta.id"
+    label 'process_low'
```
[the remark this answers is not in the document]
Comment (FriederikeHanssen) [reply modules/imputeme/vcftoprs/main.nf:10]: FYI: The amount of resources requested with `low`, `medium`, `high` are specified here: https://github.com/nf-core/tools/blob/b48bce26b7da46cb71e4a8f78b68d9ceac579ca6/nf_core/pipeline-template/conf/base.config#L29-L46

[the remark this answers is not in the document]
Comment (lassefolkersen) [reply modules/imputeme/vcftoprs/main.nf:10]: Ok, thanks - low it is then

File: modules/imputeme/vcftoprs/meta.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      tool_dev_url: https://github.com/lassefolkersen/impute-me
+      doi: "https://doi.org/10.3389/fgene.2020.00578"
+      licence: LGPL3
+
+input:
+  - meta:
+      type: map
+      description: |
+        Groovy Map containing sample information
+        e.g. [ vcf:'test', single_end:false ]
```
[the remark this answers is not in the document]
Comment (lassefolkersen) [reply modules/imputeme/vcftoprs/meta.yml:19]: Sorry, I think I just said yes to this, @pontus - but didn't actually understand. Many of the other modules (all I looked in, randomly, mosdepth, fastqc), also has this `type: map` entry. With the save vcf: test part. So I'm in doubt what I can actually change. Is it just free text description of overall input, e.g. change from
```
input:
-meta:
description: |
        Groovy Map containing sample information
        e.g. [ vcf:'test', single_end:false ]
```
To this:

```
input:
-meta:
description: |
        The only required input is a vcf file (can be either hg38 or hg19, must be from whole genome sequencing)
        e.g. [ vcf:'test' ]
```

I guess my question is - how specific can I be and how much can I change in, without breaking the boilerplate code?

**Comment by lassefolkersen** [comment]
Now there's a new release, v1.0.7, that is on biocontainers (as docker), and in the process of building as singularity. Olivier Sallou from biocontainers [said](https://github.com/BioContainers/containers/pull/427#issuecomment-958927687) that it'd also be available as singularity soon. Will wait for that, and then test if I got the singularity link right (now it's failing, probably because it's not ready)

File: modules/imputeme/vcftoprs/meta.yml (hunk 2 above)
[the remark this answers is not in the document]
Comment (pontus) [reply modules/imputeme/vcftoprs/meta.yml:19]: No worries, if you can't come up with anything you think is more descriptive, it's fine leaving as is.

**Comment by lassefolkersen** [comment]
Ok - I've filled in all the check-marks now, and it's working with singularity, pulling from biocontainers. I'm ready for review 😄 

Two known exceptions: I couldn't get it to work BioConda and the linting fails because I had to make a custom version export. It's basically just saying that the `${getProcessName(task.process)}:` and ` ${getSoftwareName(task.process)}:` are missing, but at least the version export is dynamic enough (from the version function).

**Review by pontus — APPROVED** [review]
👍

**Pull request merged**